ab16ad33ba / https://github.com/vim/vim/pull/9422 "SIGTSTP is not handled"
caused hangs on ^Z

other changes were stacked including https://github.com/vim/vim/pull/9602

https://github.com/vim/vim/pull/9854 fixed this on Android; widen this to
include OpenBSD which fixes it here too

Index: src/os_unix.c
--- src/os_unix.c.orig
+++ src/os_unix.c
@@ -887,7 +887,7 @@ sig_tstp SIGDEFARG(sigarg)
     else
 	got_tstp = TRUE;
 
-#ifndef __ANDROID__
+#if !defined(__ANDROID__) && !defined(__OpenBSD__)
     // this is not required on all systems
     signal(SIGTSTP, (RETSIGTYPE (*)())sig_tstp);
 #endif
